compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -1440,6 +1440,222 @@ LogicalResult ArgCompareOp::generateScalarImplementation(OpBuilder &b,
+      int64_t size = (cast<IntegerAttr>(sizeAttr)).getInt();
+      resultValShape.push_back(size);
+      resultIdxShape.push_back(size);
+      continue;
+    }
+
+    resultValShape.push_back(ShapedType::kDynamic);
+    resultIdxShape.push_back(ShapedType::kDynamic);
+  }
+
+  auto sliceValResultType =
+      RankedTensorType::get(resultValShape, initValType.getElementType(),
+                            cast<RankedTensorType>(initValType).getEncoding());
+  auto sliceIdxResultType =
+      RankedTensorType::get(resultIdxShape, initIdxType.getElementType(),
+                            cast<RankedTensorType>(initIdxType).getEncoding());
+
+  auto initValSlice = tensor::ExtractSliceOp::create(
+      b, loc, sliceValResultType, init[0], initOffsets, initSizes, initStrides);
+  tiledOperands.push_back(initValSlice.getResult());
+  slices.push_back(initValSlice);
+
+  auto initIdxSlice = tensor::ExtractSliceOp::create(
+      b, loc, sliceIdxResultType, init[1], initOffsets, initSizes, initStrides);
+  tiledOperands.push_back(initIdxSlice.getResult());
+  slices.push_back(initIdxSlice);
+
+  // Create index_base for this chunk.
+  Value tileIndex =
+      getValueOrCreateConstantIndexOp(b, loc, splitReductionIvs[0]);
+  Value tileSize = getValueOrCreateConstantIndexOp(b, loc, sizes[reductionDim]);
+  Value tileStartIndex = arith::MulIOp::create(b, loc, tileIndex, tileSize);
+
+  Value newIndexBase;
+  if (Value globalIndexBase = getIndexBase()) {
+    // Add chunk start to existing index_base.
+    newIndexBase =
+        arith::AddIOp::create(b, loc, globalIndexBase, tileStartIndex);
+  } else {
+    // Use chunk start as index_base.
+    newIndexBase = tileStartIndex;
+  }
+
+  SmallVector<Type, 2> resultTypes;
+  if (hasPureTensorSemantics()) {
+    resultTypes = {initValSlice->getResult(0).getType(),
+                   initIdxSlice->getResult(0).getType()};
+  }
+
+  // Create the tiled operation with adjusted index_base.
+  SmallVector<Value> operands = std::move(tiledOperands);
+  operands.push_back(newIndexBase);
+  Operation *tiledArgmaxOp =
+      b.create<ArgCompareOp>(loc, /*results=*/resultTypes,
+                             /*inputs=*/operands[0],
+                             /*outputs=*/ValueRange{operands[1], operands[2]},
+                             /*indexBase=*/operands[3],
+                             /*dimension=*/reductionDim);
+
+  // Copy the region.
+  Region &targetRegion = tiledArgmaxOp->getRegion(0);
+  Region &sourceRegion = getRegion();
+  targetRegion.takeBody(sourceRegion);
+
+  return TilingResult{
+      {tiledArgmaxOp}, SmallVector<Value>(tiledArgmaxOp->getResults()), slices};
+}
+
+FailureOr<MergeResult>
+ArgCompareOp::mergeReductions(OpBuilder &b, Location loc,
+                              ValueRange partialReduce,
+                              const llvm::SetVector<unsigned> &reductionDims) {
+  SmallVector<int64_t> mergeReductionDims =
+      llvm::to_vector_of<int64_t>(reductionDims);
```

Comment:
```suggestion
  auto mergeReductionDims =
      llvm::to_vector_of<int64_t>(reductionDims);
```
See https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable